enable_location.sh

```diff
@@ -92,23 +92,38 @@ location ${LOCATION} {
 
     include  ${NAXSI_LOCATION_RULES}/*.rules ;
 
-    set \$_request \$request;
-    if (\$_request ~ (.*)email=[^&+]*(.*)) {
-        set \$_request \$1email=****\$2;
-    }
-    set \$_http_referer \$http_referer;
-    if (\$_http_referer ~ (.*)email=[^&+]*(.*)) {
-        set \$_http_referer \$1email=****\$2;
-    }
-
-    set \$backend_upstream "\$proxy_address";
-    proxy_pass \$backend_upstream;
-    proxy_redirect  off;
-    proxy_intercept_errors on;
-    proxy_set_header X-Forwarded-For \$proxy_add_x_forwarded_for;
-    proxy_set_header X-Forwarded-Proto \$scheme;
-    proxy_set_header Host \$host:\$server_port;
-    proxy_set_header X-Real-IP \$remote_addr;
+    # Include the Nginx proxy_pass and headers - we want to use these again.
+    include /etc/nginx/conf/nginx_proxy_config.conf;
 
 }
 EOF_LOCATION_CONF
+
+# If Static Caching is enabled, add Cache Config for the Server Block...
+if [ -n "${PROXY_STATIC_CACHING:-}" ]; then
+cat >> /etc/nginx/conf/locations/${LOCATION_ID}.conf <<- EOF_SERVERCACHE_CONF
+
+# Allow Nginx to cache static assets - follow the same proxy config as above.
+location ~* ^${LOCATION}(.+\.(jpg|jpeg|gif|png|svg|ico|css|bmp|js|html|htm|ttf|otf|eot|woff|woff2)$ {
+    proxy_cache staticcache;
+    proxy_cache_bypass $http_cache_control; # Support client "Cache-Control: no-cache" directive
+    add_header X-Proxy-Cache $upstream_cache_status; # Hit or Miss
+
+    # Nginx cache to ignore Node.js "Cache-Control: public, max-age=0"
+    proxy_ignore_headers Cache-Control;
+    proxy_hide_header Cache-Control;
+    add_header Cache-Control "public";
+    expires 60m; # "Cache-Control: max-age=3600" tells client to cache for 60 minutes
```

Comment:
Do we have any cache-busting in place on frontend releases to prevent stale assets being served? This may be an acceptable time frame for hanging on to stale content, but probably worth a discussion with FED devs.

Reply:
I believe that cache busting is in the backlog, so for now, we'll have to suffice with short cache times. If they believe that 1 hour is too long, we could shorten it to something like 10 minutes.

Reply:
Could we add in a [cache purge](https://docs.nginx.com/nginx/admin-guide/content-cache/content-caching/#purging-content-from-the-cache) option? Maybe not in this PR, the access restriction stuff looks a bit involved...

Reply:
The purge feature is [exclusive to Nginx Plus](https://www.nginx.com/products/nginx/caching/#purging), the paid version of Nginx. So we wouldn't be able to use this.

Reply:
Ooo I missed that! I was wondering why I hadn't come across it before 😸